open_humans/views.py

```diff
@@ -283,15 +280,36 @@ def get_context_data(self, **kwargs):
         return context
 
 
-class DataRetrievalTaskDeleteView(PrivateMixin, DeleteView):
+class SourceDataFilesDeleteView(PrivateMixin, DeleteView):
     """
-    Let the user delete a dataset.
+    Let the user delete all datafiles for a source.
     """
+    template_name = 'member/my-member-source-data-files-delete.html'
     success_url = reverse_lazy('my-member-research-data')
 
     def get_queryset(self):
-        return DataRetrievalTask.objects.filter(user=self.request.user)
+        source = self.kwargs['source']
+
+        data_file_model = app_name_to_data_file_model(source)
+
+        return data_file_model.objects.filter(
+            user_data__user=self.request.user)
+
+    def get_object(self, queryset=None):
```

Comment:
not really a hack, get_queryset is designed to be overridden to get the queryset for deletion; the only quirk is that DeleteView expects to then filter the queryset by the pk, which is done in get_object (it expects one object). since we pre-filter in get_queryset we just pass that through. it is probably more readable to just get the queryset in get_object though! that way we only override one method.